features/dd-sdk-android-trace-otel/src/test/kotlin/com/datadog/android/trace/opentelemetry/OtelTracerBuilderProviderTest.kt

```diff
@@ -151,7 +153,24 @@ internal class OtelTracerBuilderProviderTest {
     }
 
     @Test
-    fun `M use a NoOpOtelWriter W build { TracingFeature not enabled }`() {
+    fun `M log a user error W build { TracingFeature not implementing InternalCoreTracerWriterProvider }`() {
+        // GIVEN
+        whenever(mockTracingFeatureScope.unwrap<Feature>()) doReturn mock()
+
+        // WHEN
+        val tracer = testedOtelTracerProviderBuilder.build()
+
+        // THEN
+        assertThat(tracer).isNotNull
+        mockInternalLogger.verifyLog(
+            InternalLogger.Level.ERROR,
+            InternalLogger.Target.USER,
+            OtelTracerProvider.TRACING_NOT_ENABLED_ERROR_MESSAGE
```

Comment:
how then we are going distinguish between the cases when Tracing is not enabled vs there is a mistake from our side by not implementing the necessary interface? Shouldn't we introduce different errors for these cases?

Reply:
good point, it makes sense in this case I guess to have different messages. Maybe we should also change the `target` in this case to `MAINTAINER` ?

Reply:
for the case when checking internal interface - yes, for the case when checking Feature - it should stay with User target.